detection_rules/exception.py

```diff
@@ -0,0 +1,143 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+"""Rule exceptions data."""
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Literal, Optional, Union
+
+from marshmallow import validates_schema, ValidationError
+
+from .mixins import MarshmallowDataclassMixin
+from .schemas import definitions
+
+
+# https://www.elastic.co/guide/en/security/current/exceptions-api-overview.html
+
+@dataclass(frozen=True)
+class ExceptionMeta(MarshmallowDataclassMixin):
+    """Data stored in an exception's [metadata] section of TOML."""
+    creation_date: definitions.Date
+    rule_id: definitions.UUIDString
+    rule_name: str
+    updated_date: definitions.Date
+
+    # Optional fields
+    deprecation_date: Optional[definitions.Date]
+    comments: Optional[str]
+    maturity: Optional[definitions.Maturity]
+
+
+@dataclass(frozen=True)
+class BaseExceptionItemEntry(MarshmallowDataclassMixin):
+    field: str
+    type: definitions.ExceptionEntryType
+
+
+@dataclass(frozen=True)
+class NestedExceptionItemEntry(BaseExceptionItemEntry, MarshmallowDataclassMixin):
+    entries: List['ExceptionItemEntry']
+
+    @validates_schema
+    def validate_nested_entry(self, data: dict, **kwargs):
+        if data.get('list') is not None:
+            raise ValidationError('Nested entries cannot define a list')
+
+
+@dataclass(frozen=True)
+class ExceptionItemEntry(BaseExceptionItemEntry, MarshmallowDataclassMixin):
+    @dataclass(frozen=True)
+    class ListObject:
+        id: definitions.UUIDString
+        type: definitions.EsDataTypes
+
+    list: Optional[ListObject]
+    operator: definitions.ExceptionEntryOperator
+    value: Optional[Union[str, List[str]]]
```

Comment:
How do we plan to validate these fields? https://github.com/elastic/detection-rules/issues/3536

Reply:
We may just want to leave this to the community, but if we do build in support for this, such as an external_utils, we should do it in a future PR